R/check_result.R

```diff
@@ -25,11 +29,17 @@
 #' \dontrun{grading_demo()}
 check_result <- function(
   ...,
-  correct = "{paste0(random_praise(), if (nchar(message) > 0) \" \", message)}",
-  incorrect = "{paste0(message, if(nchar(message) > 0) \" \", random_encourage())}",
-  grader_args = list(), # provided by `grade_learnr`
-  learnr_args = list() # provided by `grade_learnr`
+  correct = NULL,
+  incorrect = NULL,
+  grader_args = list(),
+  learnr_args = list(),
+  glue_correct = getOption("gradethis_glue_correct"),
+  glue_incorrect = getOption("gradethis_glue_incorrect")
 ) {
+  # convert NULL correct/incorrect strings to "" to work with glue
+  if (is.null(correct)) {correct <- ""}      # nolint
```

Comment:
rather than manually upgrading all string elements, we should make a helper function that accepts a named list and upgrades all `NULL` values to `""`.

Use it down below on the list of `glue::glue_data`

```R
glue::glue_data(
  message_list(
    .is_match = found_match,
    .is_correct = final_result$correct,
    # etc
  )
)
```

Reply:
should be resolved in #57 6e52ae9935ddc5625181935e728eca30ff1cb9be

Reply:
remove these two lines as `glue_message` covers them